traffic_portal/app/src/common/modules/table/servers/table.servers.tpl.html

```diff
@@ -23,31 +23,47 @@
             <li class="active">Servers</li>
         </ol>
         <div class="pull-right">
-            <a class="btn btn-primary" name="createServersButton" title="Create Server" href="#!/servers/new"><i class="fa fa-plus"></i></a>
-            <button class="btn btn-default" title="Refresh" ng-click="refresh()"><i class="fa fa-refresh"></i></button>
-            <div id="toggleColumns" class="btn-group" role="group" title="Select Table Columns" uib-dropdown is-open="columnSettings.isopen">
-                <button type="button" class="btn btn-default dropdown-toggle" uib-dropdown-toggle aria-haspopup="true" aria-expanded="false">
-                    <i class="fa fa-columns"></i>&nbsp;
-                    <span class="caret"></span>
-                </button>
-                <menu ng-click="$event.stopPropagation()" class="column-settings dropdown-menu-right dropdown-menu" uib-dropdown-menu>
-                    <li role="menuitem" ng-repeat="c in gridOptions.columnApi.getAllColumns() | orderBy:'colDef.headerName'">
-                        <div class="checkbox">
-                            <label><input type="checkbox" ng-checked="c.isVisible()" ng-click="toggleVisibility(c.colId)">{{::c.colDef.headerName}}</label>
-                        </div>
-                    </li>
-                </menu>
-            </div>
-            <div class="btn-group" role="group" uib-dropdown is-open="more.isopen">
-                <button type="button" class="btn btn-default dropdown-toggle" uib-dropdown-toggle aria-haspopup="true" aria-expanded="false">
-                    More&nbsp;
-                    <span class="caret"></span>
-                </button>
-                <ul class="dropdown-menu-right dropdown-menu" uib-dropdown-menu>
-                    <li role="menuitem"><a ng-click="confirmCDNQueueServerUpdates(null)">Queue CDN Server Updates</a></li>
-                    <li role="menuitem"><a ng-click="confirmCDNClearServerUpdates(null)">Clear CDN Server Updates</a></li>
-                    <li role="menuitem"><button class="menu-item-button" type="button" ng-click="exportCSV()">Export CSV</button></li>
-                </ul>
+            <div class="form-inline">
+                <div class="input-group text-input">
+                    <span class="input-group-btn">
+                        <label for="quickSearch" class="btn btn-default"><i class="fa fa-search"></i></label>
+                    </span>
+                    <input id="quickSearch" type="text" class="form-control" placeholder="Quick search..." ng-model="quickSearch" ng-change="onQuickSearchChanged()"/>
```

Comment:
Inputs without textual labels should have `aria-label` equivalents that describe what the input is (e.g. "quick search servers table"). Alternatively/Additionally, you could make `div.form-inline` into `form.form-inline[role=search]`, which will cause screen readers to announce it as a search form, which makes it more clear what's going to happen on submit (although we're capturing the `change` event of the input itself not the `submit` event of the form, so it's possibly not that helpful).

This could also be an `input[type=search]` instead of `input[type=text]` - the two are functionally identical, but browsers may choose input suggestions differently between the two. that's not really a big deal.

Reply:
I'm also generally wary of `inputs` with no `name`, but again since there is no owning `form` - and it would never be actually submitted in any case - that's probably not a big deal.

Reply:
i think i got this all addressed.